presto-kinesis/src/main/java/io/prestosql/plugin/kinesis/KinesisSessionProperties.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.kinesis;
+
+import com.google.common.collect.ImmutableList;
+import com.google.inject.Inject;
+import io.prestosql.spi.connector.ConnectorSession;
+import io.prestosql.spi.session.PropertyMetadata;
+
+import java.text.ParsePosition;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.List;
+import java.util.TimeZone;
+
+import static io.prestosql.spi.session.PropertyMetadata.booleanProperty;
+import static io.prestosql.spi.session.PropertyMetadata.integerProperty;
+import static io.prestosql.spi.session.PropertyMetadata.stringProperty;
+
+public final class KinesisSessionProperties
+{
+    public static final String PRESTO_TIMESTAMP_FORMAT = "yyyy-MM-dd HH:mm:ss.SSS";
+    public static final String UNSET_TIMESTAMP = "2000-01-01 00:00:00.000";
+
+    public static final String CHECKPOINT_ENABLED = "checkpoint-enabled"; // boolean
+    public static final String ITERATION_NUMBER = "iteration_number"; // int
+    public static final String CHECKPOINT_LOGICAL_NAME = "checkpoint_logical_name"; // string
+    public static final String MAX_BATCHES = "max_batches"; // int
+    public static final String BATCH_SIZE = "batch_size"; // int
+    public static final String ITER_FROM_TIMESTAMP = "iter_from_timestamp"; // boolean
```

Comment:
The name `iter_` doesn't seem very user friendly. How about naming these something like

* `start_from_timestamp`
* `starting_offset_seconds`
* `starting_timestamp`

Reply:
changed